admin/includes/classes/order.php

```diff
@@ -0,0 +1,187 @@
+
+      $this->billing = array('name' => $order->fields['billing_name'],
+                             'company' => $order->fields['billing_company'],
+                             'street_address' => $order->fields['billing_street_address'],
+                             'suburb' => $order->fields['billing_suburb'],
+                             'city' => $order->fields['billing_city'],
+                             'postcode' => $order->fields['billing_postcode'],
+                             'state' => $order->fields['billing_state'],
+                             'country' => $order->fields['billing_country'],
+                             'format_id' => $order->fields['billing_address_format_id']);
+
+      $index = 0;
+      $orders_products = $db->Execute("select orders_products_id, products_id, products_name, products_model,
+                                              products_price, products_tax, products_quantity,
+                                              final_price, onetime_charges,
+                                              product_is_free
+                                       from " . TABLE_ORDERS_PRODUCTS . "
+                                       where orders_id = '" . (int)$order_id . "'
+                                       order by orders_products_id");
+
+      while (!$orders_products->EOF) {
+// convert quantity to proper decimals - account history
+          if (QUANTITY_DECIMALS != 0) {
+            $fix_qty = $orders_products->fields['products_quantity'];
+            switch (true) {
+            case (!strstr($fix_qty, '.')):
+              $new_qty = $fix_qty;
+              break;
+            default:
+              $new_qty = preg_replace('/[0]+$/', '', $orders_products->fields['products_quantity']);
+              break;
+            }
+          } else {
+            $new_qty = $orders_products->fields['products_quantity'];
+          }
+
+          $new_qty = round($new_qty, QUANTITY_DECIMALS);
+
+          if ($new_qty == (int)$new_qty) {
+            $new_qty = (int)$new_qty;
+          }
+
+        $this->products[$index] = array('qty' => $new_qty,
+                                        'id' => $orders_products->fields['products_id'],
+                                        'name' => $orders_products->fields['products_name'],
+                                        'model' => $orders_products->fields['products_model'],
+                                        'tax' => $orders_products->fields['products_tax'],
+                                        'price' => $orders_products->fields['products_price'],
+                                        'onetime_charges' => $orders_products->fields['onetime_charges'],
+                                        'final_price' => $orders_products->fields['final_price'],
+                                        'product_is_free' => $orders_products->fields['product_is_free']);
+
+
+      if( STOCK_SBA_DISPLAY_CUSTOMID == 'true') {
```

Comment:
Besides that this can go into an observer that already exists, it also assumes only that a single customid is assigned, or rather doesn't address the use of multiple customid's.  Also, in development, the aspect of replacing the model with the customid had been discussed with the original developer (potteryhouse) (reduced many of the possible needs for code modification; however, because of the original intent (people for whom this was primarily developed), I won't be incorporating such a change without adding an additional switch to where the admin chooses to replace the model number with the customid.  This action also could/would be completed earlier in the process.  There are those that still want to have both the model and the customid displayed/available.

Good to see though that someone else wants this "result".